### PR TITLE
app: fix possible integer overflow in net_box.c

### PR DESCRIPTION
**Pull request by ImeevMA**
The netbox_transport_send_and_recv() function takes a part of the data buffer size as a parameter, and it is possible that the resulting buffer size could become larger than SIZE_MAX, resulting in an integer overflow and a segmentation fault.

Closes #tarantool/security#121

NO_DOC=bugfix

### CODE REVIEW

File: src/box/lua/net_box.c
Comment (locker) [inline src/box/lua/net_box.c:1168]: Nit:
 - It's not "data buffer size" that is too large. It's "response size".
 - Let's decode the length into `uint64_t` because `mp_decode_uint` returns `uint64_t`.
 - Let's put `len` to the left to emphasize that we check `len`, not `size`.

```C
uint64_t len = mp_decode_uint(&rpos);
size_t size = rpos - bufpos;
if (len > SIZE_MAX - size) {
        box_error_raise(ER_NO_CONNECTION, "Response size is too large");
        return -1;
}
```

File: changelogs/unreleased/ghs-121-too-big-buffer-size.md
Comment (locker) [inline changelogs/unreleased/ghs-121-too-big-buffer-size.md:3]: - **in** net.box.
- We never used the "app" section in changelogs. I'm not sure what section we should use for net.box, but I'd use "bugfix/core".

**Review by locker — CHANGES_REQUESTED** [review]

File: src/box/lua/net_box.c
Comment (ImeevMA) [reply src/box/lua/net_box.c]: Fixed.

File: changelogs/unreleased/ghs-121-too-big-buffer-size.md
Comment (ImeevMA) [reply changelogs/unreleased/ghs-121-too-big-buffer-size.md]: Fixed.

**Review by locker — APPROVED** [review]

**Comment by locker** [comment]
Waiting for review from the doc team.

File: changelogs/unreleased/ghs-121-too-big-buffer-size.md
Comment (andreyaksenov) [inline changelogs/unreleased/ghs-121-too-big-buffer-size.md:3]: A couple of minor suggestions:
```
* Fixed an integer overflow issue in `net.box` (ghs-121).
```

**Review by andreyaksenov — APPROVED** [review]

File: changelogs/unreleased/ghs-121-too-big-buffer-size.md
Comment (ImeevMA) [reply changelogs/unreleased/ghs-121-too-big-buffer-size.md]: Fixed.

**Pull request merged**
